Fixes dead symlink bug in `record_artifacts_as_dict`

lukpueh opened the pull request:
File existence is only checked on the first level of the file
system traversal. If we recurse in a directory using os.walk,
we might also record symlinks to non-existent files, which would
crash once we try reading the file.

This commit excludes non-existent files in our `os.walk` loop
and adapts the related unittest.

Commit:
Fixes dead symlink bug in `record_artifacts_as_dict`

File existence is only checked on the first level of the file
system traversal. If we recurse in a directory using os.walk,
we might also record symlinks to non-existent files, which would
crash once we try reading the file.

This commit excludes non-existent files in our `os.walk` loop
and adapts the related unittest.

SantiagoTorres commented:
should we `log.warn` on the else maybe? 

```
else:
   log.warn("file {} appears to be a broken symlink. Skipping..." ...)
```

Or so

lukpueh commented:
Ha, I just had the same thought. 👍

Commit:
Adds warning log statement for broken symlinks.

Commit:
Replaces `log.warning` with `log.warn`

In-toto's logging wrapper was changed in the now merged
output enhancement PR: 4d05c888a6f5c1046621cb253c8ecf87086bc121

SantiagoTorres commented:
LGTM 💯
